src/test/java/uk/nhs/hee/tis/trainee/credentials/service/VerificationServiceTest.java

```diff
@@ -0,0 +1,181 @@
+    service = new VerificationService(cachingDelegate, properties);
+  }
+
+  @Test
+  void shouldCacheIdentityDataWhenStartingVerification() {
+    IdentityDataDto dto = new IdentityDataDto("Anthony", "Gilliam", LocalDate.now());
+
+    URI uri = service.startIdentityVerification(dto, null);
+
+    ArgumentCaptor<UUID> uuidCaptor = ArgumentCaptor.forClass(UUID.class);
+    verify(cachingDelegate).cacheIdentityData(uuidCaptor.capture(), eq(dto));
+
+    String uuid = uuidCaptor.getValue().toString();
+    Map<String, String> queryParams = splitQueryParams(uri);
+    assertThat("Unexpected cache key.", uuid, is(queryParams.get("nonce")));
+  }
+
+  @Test
+  void shouldCacheClientStateWhenStartingVerification() {
+    URI uri = service.startIdentityVerification(null, "some-client-state");
+
+    ArgumentCaptor<UUID> uuidCaptor = ArgumentCaptor.forClass(UUID.class);
+    verify(cachingDelegate).cacheClientState(uuidCaptor.capture(), eq("some-client-state"));
+
+    String uuid = uuidCaptor.getValue().toString();
+    Map<String, String> queryParams = splitQueryParams(uri);
+    assertThat("Unexpected cache key.", uuid, is(queryParams.get("state")));
+  }
+
+  @Test
+  void shouldCacheCodeVerifierWhenStartingVerification() {
+    URI uri = service.startIdentityVerification(null, null);
+
+    ArgumentCaptor<UUID> uuidCaptor = ArgumentCaptor.forClass(UUID.class);
+    verify(cachingDelegate).cacheCodeVerifier(uuidCaptor.capture(), any());
+
+    String uuid = uuidCaptor.getValue().toString();
+    Map<String, String> queryParams = splitQueryParams(uri);
+    assertThat("Unexpected cache key.", uuid, is(queryParams.get("state")));
+  }
+
+  @Test
+  void shouldUserAuthorizeEndpointWhenStartingVerification() {
+    URI uri = service.startIdentityVerification(null, null);
+
+    URI relativeUri = uri.relativize(URI.create(AUTHORIZE_ENDPOINT));
+    assertThat("Unexpected relative URI.", relativeUri, is(URI.create("")));
+  }
+
+  @Test
+  void shouldIncludeNonceWhenStartingVerification() {
+    URI uri = service.startIdentityVerification(null, null);
+
+    Map<String, String> queryParams = splitQueryParams(uri);
+    String nonce = queryParams.get("nonce");
+    assertDoesNotThrow(() -> UUID.fromString(nonce));
+  }
+
+  @Test
+  void shouldIncludeStateWhenStartingVerification() {
+    URI uri = service.startIdentityVerification(null, null);
+
+    Map<String, String> queryParams = splitQueryParams(uri);
+    String state = queryParams.get("state");
+    assertDoesNotThrow(() -> UUID.fromString(state));
+  }
+
+  @Test
+  void shouldIncludeCodeChallengeMethodWhenStartingVerification() {
+    URI uri = service.startIdentityVerification(null, null);
+
+    Map<String, String> queryParams = splitQueryParams(uri);
+    String codeChallengeMethod = queryParams.get("code_challenge_method");
+    assertThat("Unexpected query parameter value.", codeChallengeMethod, is("S256"));
```

Comment:
Just going down a bit of a rabbit-hole on the various `?`S256 algorithms 🐰 . But I see it must be S256 for this.